src/qa/qa.py

```diff
@@ -17,15 +18,41 @@ def get_metric(
     ) -> Union[float, int, bool]:
         pass
 
+class QaTestRegistry:
+    registry = {}
 
-class LLMTestSuite:
-    def __init__(
-        self,
-        tests: List[LLMQaTest],
-        prompts: List[str],
-        ground_truths: List[str],
-        model_preds: List[str],
-    ) -> None:
+    @classmethod
+    def register(cls, *names):
+        def inner_wrapper(wrapped_class):
+            for name in names:
+                cls.registry[name] = wrapped_class
+            return wrapped_class
+        return inner_wrapper
+
+    @classmethod
+    def create_test(cls, test_name: str):
+        if test_name in cls.registry:
+            return cls.registry[test_name]()
+        else:
+            raise ValueError(f"Test not found: {test_name}")
+
+    @classmethod
+    def from_string_list(cls, string_list: List[str]) -> List[LLMQaTest]:
+        tests = []
+        for test_name in string_list:
+            tests.append(TestRegistry.create_test(test_name))
+        return tests
+
+    @classmethod 
+    def create_test(cls, test_name: str) -> List[LLMQaTest]:
```

Comment:
method name conflict; made it more clear

```suggestion
    def create_tests_from_list(cls, test_name: str) -> List[LLMQaTest]:
```